app/models/note.rb

```diff
@@ -14,6 +14,7 @@ class Note
 
   # Validations
   validates :created_by_id, :full_text, presence: true
+#  validate :archived_state, if: self.patient.archived? # how do I access the associated patient
```

Comment:
`note` is `embedded_in :patient`, IIRC? You can access the parent patient object with `patient` -- ex `@note.patient` should return the appropriate patient. So I would guess that `validate :archived_state, if: 'patient.archived?' would work. I didn't try this in a REPL though so if I'm wrong I will say something nice about kale as penance.

Reply:
doesn't quite work. I get this error:

`NameError: undefined local variable or method `patient' for Note:Class`